Panic while running the bank app on resolveWriteIntentError due to nil pointer dereference

[full stack trace](https://gist.githubusercontent.com/BramGruneir/a574350ef1c49cdcc37d/raw/gistfile1.txt)
```
I0624 14:58:11.773672   22542 store.go:1356] on Get: txn "bank.go:154" id=d9de0973 key="000000005" pri=99.96220432 iso=SERIALIZABLE stat=PENDING epo=3 ts=1435172276.927472353,3 orig=1435172276.927472353,3 max=1435172274.682641506,0 failed to push "bank.go:154" id=2e92ce96 key="000000008" pri=99.96220437 iso=SERIALIZABLE stat=PENDING epo=3 ts=1435172276.927472353,2 orig=1435172274.215435743,1 max=1435172274.215435743,1
I0624 14:58:11.778768   22542 store.go:1356] on Get: txn "bank.go:154" id=aeccb05a key="000000006" pri=99.96220432 iso=SERIALIZABLE stat=PENDING epo=5 ts=1435172276.927472353,2 orig=1435172276.927472353,2 max=1435172276.927472353,2 failed to push "bank.go:154" id=2e92ce96 key="000000008" pri=99.96220437 iso=SERIALIZABLE stat=PENDING epo=3 ts=1435172276.927472353,2 orig=1435172274.215435743,1 max=1435172274.215435743,1
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4652a65]

goroutine 473487 [running]:
github.com/cockroachdb/cockroach/storage.(*Store).resolveWriteIntentError(0xc20815e000, 0x5439b80, 0xc20986cf90, 0xc20cd2ee80, 0xc208160090, 0x5447b68, 0xc208dfc100, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1385 +0x1055
github.com/cockroachdb/cockroach/storage.func·042(0xc20ac21570, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1262 +0x2f3
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc2088060f0, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc20ac217b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc20986cf90, 0x5447b68, 0xc208dfc100, 0x5447bc0, 0xc2090ea070, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc20986cf30, 0x5447b68, 0xc208dfc100, 0x5447bc0, 0xc2090ea070, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5447b68, 0xc208dfc100, 0x5447bc0, 0xc2090ea070, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Get(0xc20806e500, 0xc208dfc100, 0xc2090ea070, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:565 +0xb1
reflect.Value.call(0x49d7280, 0x4aebac0, 0x13, 0x4b16390, 0x4, 0xc20ac21f28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7280, 0x4aebac0, 0x13, 0xc20ac21f28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061800, 0xc20c347680, 0x4ae75e0, 0xc208dfc100, 0x16, 0x4add300, 0xc2090ea070, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 1 [select, 4 minutes]:
github.com/cockroachdb/cockroach/server/cli.runStart(0x4f32160, 0xc208040cd0, 0x0, 0x5)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/cli/start.go:140 +0xdcc
github.com/spf13/cobra.(*Command).execute(0x4f32160, 0xc208040c80, 0x5, 0x5, 0x0, 0x0)
	/Users/bram/go/src/github.com/spf13/cobra/command.go:495 +0x65c
github.com/spf13/cobra.(*Command).Execute(0x4f30420, 0x0, 0x0)
	/Users/bram/go/src/github.com/spf13/cobra/command.go:560 +0x18d
github.com/cockroachdb/cockroach/server/cli.Run(0xc20800a010, 0x6, 0x6, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/cli/cli.go:118 +0x88
main.main()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/main.go:48 +0x29d

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x4f704e0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1034 +0x75
created by github.com/cockroachdb/cockroach/util/log.init·1
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:604 +0xcc

goroutine 17 [syscall, 5 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 7 [syscall, 5 minutes]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 82 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.func·013()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/queue.go:207 +0x2cb
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 10 [select]:
github.com/cockroachdb/cockroach/rpc.(*RemoteClockMonitor).MonitorRemoteOffsets(0xc20801d680, 0xc208042840)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/clock_offset.go:158 +0x637
github.com/cockroachdb/cockroach/server.func·010()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/server.go:106 +0x3a
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 11 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc20807c510)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:261 +0x8f9
github.com/cockroachdb/cockroach/kv.*TxnCoordSender.(github.com/cockroachdb/cockroach/kv.startStats)·fm()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:248 +0x27
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 12 [select]:
github.com/cockroachdb/cockroach/gossip.func·021()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/gossip/server.go:156 +0x139
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 13 [select, 4 minutes]:
github.com/cockroachdb/cockroach/gossip.func·004()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:431 +0x1f2
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 14 [select]:
github.com/cockroachdb/cockroach/gossip.func·005()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/gossip/gossip.go:455 +0x1ab
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 476316 [select]:
github.com/cockroachdb/cockroach/rpc.sendOne(0xc2081f6000, 0x12a05f200, 0xc20c5634c0, 0x14, 0x4af1360, 0xc20b860780, 0x4adda00, 0xc20ad92d90, 0xc20c7a8de0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/send.go:235 +0x9af
created by github.com/cockroachdb/cockroach/rpc.Send
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/send.go:173 +0x1578

goroutine 16 [chan receive]:
github.com/cockroachdb/cockroach/storage.ProcessStoreEvents(0x5439268, 0xc2080fc380, 0xc20810c290)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/feed.go:256 +0x61
created by github.com/cockroachdb/cockroach/server/status.(*NodeStatusMonitor).StartMonitorFeed
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/status/monitor.go:106 +0x7f

goroutine 18 [chan receive]:
github.com/cockroachdb/cockroach/server/status.ProcessNodeEvents(0x54392d8, 0xc2080fc380, 0xc20810c2a0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/status/feed.go:84 +0x61
created by github.com/cockroachdb/cockroach/server/status.(*NodeStatusMonitor).StartMonitorFeed
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/status/monitor.go:107 +0xe2

goroutine 476422 [semacquire]:
sync.(*WaitGroup).Wait(0xc20c4a7640)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/cockroachdb/cockroach/storage.(*Range).beginCmd(0xc208160090, 0xc20c65b600, 0xc208042701, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:527 +0x16b
github.com/cockroachdb/cockroach/storage.(*Range).addReadOnlyCmd(0xc208160090, 0x5439b80, 0xc20c2d3260, 0x5447b68, 0xc20c65b600, 0x5447bc0, 0xc2081fec40, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:599 +0x624
github.com/cockroachdb/cockroach/storage.(*Range).AddCmd(0xc208160090, 0x5439b80, 0xc20c2d3260, 0x5447b68, 0xc20c65b600, 0x5447bc0, 0xc2081fec40, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:510 +0x4a1
github.com/cockroachdb/cockroach/storage.func·042(0x6, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1249 +0x1e8
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20ad20300, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc20a4ef7b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc20c2d3260, 0x5447b68, 0xc20c65b600, 0x5447bc0, 0xc2081fec40, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc20c2d3200, 0x5447b68, 0xc20c65b600, 0x5447bc0, 0xc2081fec40, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5447b68, 0xc20c65b600, 0x5447bc0, 0xc2081fec40, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Get(0xc20806e500, 0xc20c65b600, 0xc2081fec40, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:565 +0xb1
reflect.Value.call(0x49d7280, 0x4aebac0, 0x13, 0x4b16390, 0x4, 0xc20a4eff28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7280, 0x4aebac0, 0x13, 0xc20a4eff28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061800, 0xc208cf6bc0, 0x4ae75e0, 0xc20c65b600, 0x16, 0x4add300, 0xc2081fec40, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 35 [runnable, locked to thread]:
github.com/cockroachdb/cockroach/storage/engine._Cfunc_DBIterDestroy(0x6200000)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/engine/:93 +0x45
github.com/cockroachdb/cockroach/storage/engine.(*rocksDBIterator).Close(0xc2081bc490)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:711 +0x2b
github.com/cockroachdb/cockroach/storage/engine.MVCCIterate(0x5437e68, 0xc208042720, 0xc20c369560, 0x12, 0x12, 0xc20c3695a0, 0x12, 0x12, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/engine/mvcc.go:1028 +0x592
github.com/cockroachdb/cockroach/storage.(*Range).Entries(0xc208160090, 0x15c19, 0x15c1a, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range_raftstorage.go:95 +0x29b
github.com/cockroachdb/cockroach/storage.(*Range).Term(0xc208160090, 0x15c19, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range_raftstorage.go:112 +0x6c
github.com/coreos/etcd/raft.(*raftLog).term(0xc20819a960, 0x15c19, 0x4f798b0)
	/Users/bram/go/src/github.com/coreos/etcd/raft/log.go:209 +0xcf
github.com/coreos/etcd/raft.(*raftLog).matchTerm(0xc20819a960, 0x15c19, 0xa, 0xc208cb9300)
	/Users/bram/go/src/github.com/coreos/etcd/raft/log.go:239 +0x32
github.com/coreos/etcd/raft.(*raftLog).maybeAppend(0xc20819a960, 0x15c19, 0xa, 0x15c18, 0x0, 0x0, 0x0, 0x15c19, 0x80)
	/Users/bram/go/src/github.com/coreos/etcd/raft/log.go:74 +0x58
github.com/coreos/etcd/raft.(*raft).handleAppendEntries(0xc2080ee600, 0x3, 0x100000001, 0x200000002, 0xa, 0xa, 0x15c19, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/coreos/etcd/raft/raft.go:657 +0x24f
github.com/coreos/etcd/raft.stepFollower(0xc2080ee600, 0x3, 0x100000001, 0x200000002, 0xa, 0xa, 0x15c19, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/coreos/etcd/raft/raft.go:628 +0x3ba
github.com/coreos/etcd/raft.(*raft).Step(0xc2080ee600, 0x3, 0x100000001, 0x200000002, 0xa, 0xa, 0x15c19, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/coreos/etcd/raft/raft.go:487 +0x259
github.com/coreos/etcd/raft.(*multiNode).run(0xc2081046c0)
	/Users/bram/go/src/github.com/coreos/etcd/raft/multinode.go:236 +0xd0b
created by github.com/coreos/etcd/raft.StartMultiNode
	/Users/bram/go/src/github.com/coreos/etcd/raft/multinode.go:56 +0xb6

goroutine 36 [select]:
github.com/coreos/etcd/raft.(*multiNode).step(0xc2081046c0, 0x54474a0, 0xc208028b58, 0x1, 0x3, 0x100000001, 0x200000002, 0xa, 0xa, 0x15c19, ...)
	/Users/bram/go/src/github.com/coreos/etcd/raft/multinode.go:407 +0x1ee
github.com/coreos/etcd/raft.(*multiNode).Step(0xc2081046c0, 0x54474a0, 0xc208028b58, 0x1, 0x3, 0x100000001, 0x200000002, 0xa, 0xa, 0x15c19, ...)
	/Users/bram/go/src/github.com/coreos/etcd/raft/multinode.go:438 +0x135
github.com/cockroachdb/cockroach/multiraft.func·005()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:521 +0xba5
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 37 [semacquire]:
sync.(*Mutex).Lock(0xc2081600e8)
	/usr/local/go/src/sync/mutex.go:66 +0xd3
sync.(*RWMutex).Lock(0xc2081600e8)
	/usr/local/go/src/sync/rwmutex.go:82 +0x33
github.com/cockroachdb/cockroach/storage.(*Range).processRaftCommand(0xc208160090, 0xc20b95fff0, 0x10, 0x15c17, 0x1, 0x100000001, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:756 +0xd7
github.com/cockroachdb/cockroach/storage.func·043()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1492 +0xa6b
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 20 [select]:
github.com/cockroachdb/cockroach/server.func·006()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:451 +0x78e
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 224 [IO wait]:
net.(*pollDesc).Wait(0xc20806d480, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20806d480, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20806d420, 0xc2081c0000, 0x1000, 0x1000, 0x0, 0x5437028, 0xc208ef4660)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20819e280, 0xc2081c0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc2081c93e0)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc2081c93e0, 0xc20c288700, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x5446c00, 0xc2081c93e0, 0x5432da8, 0x0, 0x0)
	/usr/local/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc2082041c0, 0x5446bc8, 0xc208204330, 0x0, 0x4cb4850, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).readRequestHeader(0xc2082041c0, 0xc2081c93e0, 0xc208204330, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:180 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).ReadRequestHeader(0xc2082041c0, 0xc20963cc20, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:64 +0x6e
net/rpc.(*Server).readRequestHeader(0xc2080fc200, 0x5446b88, 0xc2082041c0, 0x0, 0x0, 0xc20963cc20, 0xc20821dd00, 0x0, 0x0)
	/usr/local/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2080fc200, 0x5446b88, 0xc2082041c0, 0xc208061300, 0xc2095ef760, 0x4aa8100, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2080fc200, 0x5446b88, 0xc2082041c0)
	/usr/local/go/src/net/rpc/server.go:461 +0x6d
github.com/cockroachdb/cockroach/rpc.(*Server).serveConn(0xc208061280, 0x5442920, 0xc20819e280, 0xc2082175a0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/server.go:254 +0x95
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP(0xc208061280, 0x5446ac8, 0xc2081c2500, 0xc208168f70)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/server.go:115 +0x47d
net/http.serverHandler.ServeHTTP(0xc2080426c0, 0x5446ac8, 0xc2081c2500, 0xc208168f70)
	/usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2081c2460)
	/usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x35e

goroutine 40 [select]:
github.com/cockroachdb/cockroach/storage.func·036()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:584 +0x39e
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 41 [select]:
github.com/cockroachdb/cockroach/storage.func·037()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:604 +0x39e
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 476407 [semacquire]:
sync.(*WaitGroup).Wait(0xc20c4e1780)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/cockroachdb/cockroach/storage.(*Range).beginCmd(0xc208160090, 0xc20b7be280, 0xc209324d00, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:527 +0x16b
github.com/cockroachdb/cockroach/storage.(*Range).addWriteCmd(0xc208160090, 0x5439b80, 0xc20c31e810, 0x545a638, 0xc20b7be280, 0x545a730, 0xc208307d50, 0xc209357501, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:640 +0x14d
github.com/cockroachdb/cockroach/storage.(*Range).AddCmd(0xc208160090, 0x5439b80, 0xc20c31e810, 0x545a638, 0xc20b7be280, 0x545a730, 0xc208307d50, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:512 +0x4f6
github.com/cockroachdb/cockroach/storage.func·042(0x6, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1249 +0x1e8
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20c4e1760, 0x16, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc2093257b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc20c31e810, 0x545a638, 0xc20b7be280, 0x545a730, 0xc208307d50, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc20c31e7b0, 0x545a638, 0xc20b7be280, 0x545a730, 0xc208307d50, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x545a638, 0xc20b7be280, 0x545a730, 0xc208307d50, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).InternalPushTxn(0xc20806e500, 0xc20b7be280, 0xc208307d50, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:617 +0xb1
reflect.Value.call(0x49d7640, 0x4aebbe0, 0x13, 0x4b16390, 0x4, 0xc209325f28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7640, 0x4aebbe0, 0x13, 0xc209325f28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc20818c568, 0xc208061b00, 0xc208252380, 0x4af1360, 0xc20b7be280, 0x16, 0x4adda00, 0xc208307d50, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 50 [select]:
github.com/cockroachdb/cockroach/multiraft.func·008()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/multiraft/storage.go:145 +0x90a
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 83 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.func·013()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/queue.go:207 +0x2cb
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 84 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.func·013()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/queue.go:207 +0x2cb
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 85 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.func·013()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/queue.go:207 +0x2cb
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 86 [select, 4 minutes]:
github.com/cockroachdb/cockroach/storage.func·013()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/queue.go:207 +0x2cb
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 87 [select]:
github.com/cockroachdb/cockroach/storage.(*rangeScanner).waitAndProcess(0xc2081022a0, 0xecd1cf4c2, 0xe21bbfa30, 0x4f6ff00, 0xc2080fc1c0, 0xc208042840, 0xc20c2935f0, 0xc208160090, 0xc200000001)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:192 +0x6f5
github.com/cockroachdb/cockroach/storage.func·031(0xc208160090, 0xc20814e150)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:240 +0x71
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc20814e1b0, 0xc20b06cf00)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:233 +0x280
github.com/cockroachdb/cockroach/storage.func·032()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:242 +0x5a5
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 473893 [semacquire]:
sync.(*WaitGroup).Wait(0xc20c681840)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/cockroachdb/cockroach/storage.(*Range).beginCmd(0xc208160090, 0xc2088ee100, 0xdeaddeaddeadde01, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:527 +0x16b
github.com/cockroachdb/cockroach/storage.(*Range).addReadOnlyCmd(0xc208160090, 0x5439b80, 0xc20c8d5950, 0x5447b68, 0xc2088ee100, 0x5447bc0, 0xc20b65ad20, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:599 +0x624
github.com/cockroachdb/cockroach/storage.(*Range).AddCmd(0xc208160090, 0x5439b80, 0xc20c8d5950, 0x5447b68, 0xc2088ee100, 0x5447bc0, 0xc20b65ad20, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:510 +0x4a1
github.com/cockroachdb/cockroach/storage.func·042(0xc20c5fd570, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1249 +0x1e8
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc208435f40, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc20c5fd7b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc20c8d5950, 0x5447b68, 0xc2088ee100, 0x5447bc0, 0xc20b65ad20, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc20c8d58f0, 0x5447b68, 0xc2088ee100, 0x5447bc0, 0xc20b65ad20, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5447b68, 0xc2088ee100, 0x5447bc0, 0xc20b65ad20, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Get(0xc20806e500, 0xc2088ee100, 0xc20b65ad20, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:565 +0xb1
reflect.Value.call(0x49d7280, 0x4aebac0, 0x13, 0x4b16390, 0x4, 0xc20c5fdf28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7280, 0x4aebac0, 0x13, 0xc20c5fdf28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061800, 0xc2082821a0, 0x4ae75e0, 0xc2088ee100, 0x16, 0x4add300, 0xc20b65ad20, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 98 [select]:
github.com/cockroachdb/cockroach/rpc.(*Client).runHeartbeat(0xc2081f6000, 0xc208042840)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/client.go:243 +0x367
github.com/cockroachdb/cockroach/rpc.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/client.go:125 +0x2a7
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 114 [IO wait]:
net.(*pollDesc).Wait(0xc20806c8b0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20806c8b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20806c850, 0xc208202000, 0x1000, 0x1000, 0x0, 0x5437028, 0xc208ef4040)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2081bc080, 0xc208202000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc20813c180)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc20813c180, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x5446c00, 0xc20813c180, 0x42bd7e9, 0x0, 0x0)
	/usr/local/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc208204000, 0x5446bc8, 0xc208204170, 0x0, 0x4cb4850, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).readRequestHeader(0xc208204000, 0xc20813c180, 0xc208204170, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:180 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*serverCodec).ReadRequestHeader(0xc208204000, 0xc20c05c020, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/server.go:64 +0x6e
net/rpc.(*Server).readRequestHeader(0xc2080fc200, 0x5446b88, 0xc208204000, 0x0, 0x0, 0xc20c05c020, 0xc20817a900, 0x0, 0x0)
	/usr/local/go/src/net/rpc/server.go:575 +0x97
net/rpc.(*Server).readRequest(0xc2080fc200, 0x5446b88, 0xc208204000, 0xc208061b00, 0xc20955b780, 0x4af1360, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/rpc/server.go:542 +0xa2
net/rpc.(*Server).ServeCodec(0xc2080fc200, 0x5446b88, 0xc208204000)
	/usr/local/go/src/net/rpc/server.go:461 +0x6d
github.com/cockroachdb/cockroach/rpc.(*Server).serveConn(0xc208061280, 0x5442920, 0xc2081bc080, 0xc20818a560)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/server.go:254 +0x95
github.com/cockroachdb/cockroach/rpc.(*Server).ServeHTTP(0xc208061280, 0x5446ac8, 0xc2081ae1e0, 0xc2080732b0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/server.go:115 +0x47d
net/http.serverHandler.ServeHTTP(0xc2080426c0, 0x5446ac8, 0xc2081ae1e0, 0xc2080732b0)
	/usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2081c21e0)
	/usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x35e

goroutine 476458 [semacquire]:
sync.(*Mutex).Lock(0xc2081600e0)
	/usr/local/go/src/sync/mutex.go:66 +0xd3
github.com/cockroachdb/cockroach/storage.(*Range).redirectOnOrAcquireLeaderLease(0xc208160090, 0x13eac19920ebd504, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:371 +0x5e
github.com/cockroachdb/cockroach/storage.(*Range).addReadOnlyCmd(0xc208160090, 0x5439b80, 0xc208cb2060, 0x5453e48, 0xc20c323d40, 0x5453ea0, 0xc20ae56100, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:602 +0x688
github.com/cockroachdb/cockroach/storage.(*Range).AddCmd(0xc208160090, 0x5439b80, 0xc208cb2060, 0x5453e48, 0xc20c323d40, 0x5453ea0, 0xc20ae56100, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:510 +0x4a1
github.com/cockroachdb/cockroach/storage.func·042(0x6, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1249 +0x1e8
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20ad20a50, 0xb, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc208db57b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc208cb2060, 0x5453e48, 0xc20c323d40, 0x5453ea0, 0xc20ae56100, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc208cb2000, 0x5453e48, 0xc20c323d40, 0x5453ea0, 0xc20ae56100, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5453e48, 0xc20c323d40, 0x5453ea0, 0xc20ae56100, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Scan(0xc20806e500, 0xc20c323d40, 0xc20ae56100, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:589 +0xb1
reflect.Value.call(0x49d7a00, 0x4aebd00, 0x13, 0x4b16390, 0x4, 0xc208db5f28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7a00, 0x4aebd00, 0x13, 0xc208db5f28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061e00, 0xc208114b80, 0x4af0080, 0xc20c323d40, 0x16, 0x4ae2840, 0xc20ae56100, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 21 [select]:
github.com/cockroachdb/cockroach/server.func·007()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:522 +0x22c
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 22 [select]:
github.com/cockroachdb/cockroach/server.func·003()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:416 +0x15a
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 23 [select]:
github.com/cockroachdb/cockroach/ts.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/ts/db.go:77 +0x141
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 24 [select]:
github.com/cockroachdb/cockroach/ts.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/ts/db.go:77 +0x141
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 25 [IO wait]:
net.(*pollDesc).Wait(0xc20806cfb0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20806cfb0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20806cf50, 0x0, 0x5437028, 0xc20ad20280)
	/usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2080381a0, 0xc208089e20, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc2080381a0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc2080426c0, 0x54381f8, 0xc2080381a0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1728 +0x92
github.com/cockroachdb/cockroach/rpc.func·004()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/server.go:165 +0x53
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 26 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/rpc.func·005()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/server.go:173 +0x55
github.com/cockroachdb/cockroach/util.func·001()
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:75 +0x51
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/stopper.go:76 +0xe3

goroutine 472335 [select]:
github.com/cockroachdb/cockroach/rpc.Send(0x1, 0x1, 0x1dcd6500, 0x12a05f200, 0xc20c54dc40, 0x14, 0xc20aa3c770, 0x1, 0x1, 0xc20c54dbe0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/send.go:176 +0x12c0
github.com/cockroachdb/cockroach/kv.(*DistSender).sendRPC(0xc208061380, 0x1, 0xc20c25d550, 0x1, 0x1, 0x1, 0x545a638, 0xc20aca8780, 0x545a730, 0xc20b5fc150, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:465 +0x8ae
github.com/cockroachdb/cockroach/kv.(*DistSender).sendAttempt(0xc208061380, 0xc2081032d0, 0x545a638, 0xc20aca8780, 0x545a730, 0xc20b5fc150, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:506 +0x23c
github.com/cockroachdb/cockroach/kv.func·008(0xc200000000, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:681 +0x359
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20c54db40, 0x1b, 0xee6b280, 0x6fc23ac00, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc20b479d48, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/kv.(*DistSender).Send(0xc208061380, 0x5439b40, 0xc208028bf0, 0x545a638, 0xc20aca8780, 0x545a730, 0xc20b5fc150, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:682 +0x6b9
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).sendOne(0xc20807c510, 0x545a638, 0xc20aca8780, 0x545a730, 0xc20b5fc150, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:390 +0x273
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).sendBatch(0xc20807c510, 0xc20b34c900, 0xc20afdf400)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:535 +0x753
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).Send(0xc20807c510, 0x5439b40, 0xc208028bf0, 0x545a690, 0xc20b34c900, 0x545a6e8, 0xc20afdf400, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:315 +0x134
github.com/cockroachdb/cockroach/client.(*DB).send(0xc20806cee0, 0xc20b8e3540, 0x1, 0x1, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/client/db.go:432 +0x3e2
github.com/cockroachdb/cockroach/client.(*DB).Run(0xc20806cee0, 0xc20b364c00, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/client/db.go:390 +0x8e
github.com/cockroachdb/cockroach/storage.(*Store).resolveWriteIntentError(0xc20815e000, 0x5439b80, 0xc20c366bd0, 0xc20b8e3280, 0xc208160090, 0x5447b68, 0xc20b7a8800, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1354 +0x750
github.com/cockroachdb/cockroach/storage.func·042(0xc20b47b570, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1262 +0x2f3
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20c364eb0, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc20b47b7b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc20c366bd0, 0x5447b68, 0xc20b7a8800, 0x5447bc0, 0xc208f67d50, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc20c366b70, 0x5447b68, 0xc20b7a8800, 0x5447bc0, 0xc208f67d50, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5447b68, 0xc20b7a8800, 0x5447bc0, 0xc208f67d50, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Get(0xc20806e500, 0xc20b7a8800, 0xc208f67d50, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:565 +0xb1
reflect.Value.call(0x49d7280, 0x4aebac0, 0x13, 0x4b16390, 0x4, 0xc20b47bf28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7280, 0x4aebac0, 0x13, 0xc20b47bf28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061800, 0xc20a53eb60, 0x4ae75e0, 0xc20b7a8800, 0x16, 0x4add300, 0xc208f67d50, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 27 [IO wait]:
net.(*pollDesc).Wait(0xc20806c840, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20806c840, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20806c7e0, 0xc208206000, 0x1000, 0x1000, 0x0, 0x5437028, 0xc20ad21300)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20819e080, 0xc208206000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc208042900)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadByte(0xc208042900, 0xff4, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:196 +0x86
encoding/binary.ReadUvarint(0x5446c00, 0xc208042900, 0x4f70f20, 0x0, 0x0)
	/usr/local/go/src/encoding/binary/varint.go:110 +0x5b
github.com/cockroachdb/cockroach/rpc/codec.(*baseConn).recvProto(0xc20819c680, 0x5446cd0, 0xc20819c7d0, 0x0, 0x4cb4850, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/conn.go:91 +0x76
github.com/cockroachdb/cockroach/rpc/codec.(*clientCodec).readResponseHeader(0xc20819c680, 0xc20819c7d0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/client.go:155 +0x86
github.com/cockroachdb/cockroach/rpc/codec.(*clientCodec).ReadResponseHeader(0xc20819c680, 0xc20814e960, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/rpc/codec/client.go:80 +0x54
net/rpc.(*Client).input(0xc208042960)
	/usr/local/go/src/net/rpc/client.go:109 +0xb6
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd0

goroutine 473571 [semacquire]:
sync.(*WaitGroup).Wait(0xc20c36d6e0)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/cockroachdb/cockroach/storage.(*Range).beginCmd(0xc208160090, 0xc208dfc600, 0x1, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:527 +0x16b
github.com/cockroachdb/cockroach/storage.(*Range).addReadOnlyCmd(0xc208160090, 0x5439b80, 0xc208f1c780, 0x5447b68, 0xc208dfc600, 0x5447bc0, 0xc2090ea380, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:599 +0x624
github.com/cockroachdb/cockroach/storage.(*Range).AddCmd(0xc208160090, 0x5439b80, 0xc208f1c780, 0x5447b68, 0xc208dfc600, 0x5447bc0, 0xc2090ea380, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:510 +0x4a1
github.com/cockroachdb/cockroach/storage.func·042(0xc208ffb570, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1249 +0x1e8
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc2096ae790, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc208ffb7b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc208f1c780, 0x5447b68, 0xc208dfc600, 0x5447bc0, 0xc2090ea380, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc208f1c720, 0x5447b68, 0xc208dfc600, 0x5447bc0, 0xc2090ea380, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5447b68, 0xc208dfc600, 0x5447bc0, 0xc2090ea380, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Get(0xc20806e500, 0xc208dfc600, 0xc2090ea380, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:565 +0xb1
reflect.Value.call(0x49d7280, 0x4aebac0, 0x13, 0x4b16390, 0x4, 0xc208ffbf28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7280, 0x4aebac0, 0x13, 0xc208ffbf28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061800, 0xc208aa2d00, 0x4ae75e0, 0xc208dfc600, 0x16, 0x4add300, 0xc2090ea380, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 470088 [semacquire]:
sync.(*WaitGroup).Wait(0xc20c368d20)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/cockroachdb/cockroach/storage.(*Range).beginCmd(0xc208160090, 0xc208f49000, 0x1, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:527 +0x16b
github.com/cockroachdb/cockroach/storage.(*Range).addReadOnlyCmd(0xc208160090, 0x5439b80, 0xc208262a50, 0x5447b68, 0xc208f49000, 0x5447bc0, 0xc209300f50, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:599 +0x624
github.com/cockroachdb/cockroach/storage.(*Range).AddCmd(0xc208160090, 0x5439b80, 0xc208262a50, 0x5447b68, 0xc208f49000, 0x5447bc0, 0xc209300f50, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:510 +0x4a1
github.com/cockroachdb/cockroach/storage.func·042(0xc209f67570, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1249 +0x1e8
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc208257930, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc209f677b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc208262a50, 0x5447b68, 0xc208f49000, 0x5447bc0, 0xc209300f50, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc2082629f0, 0x5447b68, 0xc208f49000, 0x5447bc0, 0xc209300f50, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5447b68, 0xc208f49000, 0x5447bc0, 0xc209300f50, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Get(0xc20806e500, 0xc208f49000, 0xc209300f50, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:565 +0xb1
reflect.Value.call(0x49d7280, 0x4aebac0, 0x13, 0x4b16390, 0x4, 0xc209f67f28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7280, 0x4aebac0, 0x13, 0xc209f67f28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061800, 0xc209e86600, 0x4ae75e0, 0xc208f49000, 0x16, 0x4add300, 0xc209300f50, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a

goroutine 471809 [semacquire]:
sync.(*WaitGroup).Wait(0xc20c598980)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/cockroachdb/cockroach/storage.(*Range).beginCmd(0xc208160090, 0xc208440400, 0xc208042701, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:527 +0x16b
github.com/cockroachdb/cockroach/storage.(*Range).addReadOnlyCmd(0xc208160090, 0x5439b80, 0xc20a427aa0, 0x5447b68, 0xc208440400, 0x5447bc0, 0xc2081fe9a0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:599 +0x624
github.com/cockroachdb/cockroach/storage.(*Range).AddCmd(0xc208160090, 0x5439b80, 0xc20a427aa0, 0x5447b68, 0xc208440400, 0x5447bc0, 0xc2081fe9a0, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/range.go:510 +0x4a1
github.com/cockroachdb/cockroach/storage.func·042(0xc2097c9570, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1249 +0x1e8
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20b129db0, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc2097c97b0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/storage.(*Store).ExecuteCmd(0xc20815e000, 0x5439b80, 0xc20a427aa0, 0x5447b68, 0xc208440400, 0x5447bc0, 0xc2081fe9a0, 0x0, 0x0, 0x0, ...)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1287 +0x687
github.com/cockroachdb/cockroach/kv.(*LocalSender).Send(0xc20801d780, 0x5439b80, 0xc20a427a40, 0x5447b68, 0xc208440400, 0x5447bc0, 0xc2081fe9a0, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/kv/local_sender.go:156 +0x514
github.com/cockroachdb/cockroach/server.(*nodeServer).executeCmd(0xc20806e500, 0x5447b68, 0xc208440400, 0x5447bc0, 0xc2081fe9a0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:559 +0xd3
github.com/cockroachdb/cockroach/server.(*nodeServer).Get(0xc20806e500, 0xc208440400, 0xc2081fe9a0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/server/node.go:565 +0xb1
reflect.Value.call(0x49d7280, 0x4aebac0, 0x13, 0x4b16390, 0x4, 0xc2097c9f28, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/go/src/reflect/value.go:419 +0x10e5
reflect.Value.Call(0x49d7280, 0x4aebac0, 0x13, 0xc2097c9f28, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/go/src/reflect/value.go:296 +0xbc
net/rpc.(*service).call(0xc2080fc500, 0xc2080fc200, 0xc208232618, 0xc208061800, 0xc20ad1ee20, 0x4ae75e0, 0xc208440400, 0x16, 0x4add300, 0xc2081fe9a0, ...)
	/usr/local/go/src/net/rpc/server.go:382 +0x1f7
created by net/rpc.(*Server).ServeCodec
	/usr/local/go/src/net/rpc/server.go:476 +0x44a
```

I was running the bank app overnight using @tamird load balancer setup (via nginx).  Got the same panic on two different nodes this time.  I can paste the stack trace for both of them, but they look pretty similar.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4653115]

goroutine 354187 [running]:
github.com/cockroachdb/cockroach/storage.(*Store).resolveWriteIntentError(0xc2080f2180, 0x545b6a0, 0xc20a55a7b0, 0xc2098e0100, 0xc20820e000, 0x5480918, 0xc208b84800, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1385 +0x1055
github.com/cockroachdb/cockroach/storage.func·042(0x6, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1262 +0x2f3
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20aa5f900, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc20a0377b0, 0x0, ...)
...
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4653115]

goroutine 702552 [running]:
github.com/cockroachdb/cockroach/storage.(*Store).resolveWriteIntentError(0xc208170000, 0x545c610, 0xc209989b30, 0xc208679c40, 0xc208214090, 0x5480da8, 0xc208b8b300, 0x0, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1385 +0x1055
github.com/cockroachdb/cockroach/storage.func·042(0x6, 0x0, 0x0)
	/Users/bram/go/src/github.com/cockroachdb/cockroach/storage/store.go:1262 +0x2f3
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20ad1d870, 0xa, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc20b4d77b0, 0x0, ...)
...
```

```
panic: pushReply.PusheeTxn is nil: header:<timestamp:<wall_time:0 logical:0 > txn:<name:"bank.go:149" key:"000000006" id:"\345\254?<+\362O3\227Z\202,1\363\270M" priority:2147324739 isolation:SERIALIZABLE status:PENDING epoch:4 timestamp:<wall_time:1435253015670191959 logical:0 > orig_timestamp:<wall_time:1435253015670191959 logical:0 > max_timestamp:<wall_time:1435253015920191959 logical:0 > certain_nodes:<> > >
```

--- a/storage/range.go
+++ b/storage/range.go
@@ -866,7 +866,9 @@ func (r *Range) applyRaftCommand(ctx context.Context, index uint64, originNode p
 			if log.V(1) {
 				log.Infoc(ctx, "found response cache entry for %+v", args.Header().CmdID)
 			}
-			return err
+			// We successfully read from the response cache, so return whatever error
+			// was present in the cached entry (if any).
+			return reply.Header().GoError()
 		} else if ok && err != nil {
 			return newReplicaCorruptionError(
 				util.Errorf("could not read from response cache"), err)

--- a/storage/range_test.go
+++ b/storage/range_test.go
@@ -19,6 +19,7 @@ package storage
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
 	"math"
 	"reflect"
@@ -1389,10 +1390,10 @@ func TestRangeIdempotence(t *testing.T) {
 	}
 }
 
-// TestRangeResponseCacheError verifies that an error is returned to the
+// TestRangeResponseCacheReadError verifies that an error is returned to the
 // client in the event that a response cache entry is found but is not
 // decodable.
-func TestRangeResponseCacheError(t *testing.T) {
+func TestRangeResponseCacheReadError(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	tc := testContext{}
 	tc.Start(t)
@@ -1421,6 +1422,37 @@ func TestRangeResponseCacheError(t *testing.T) {
 
 	if err == nil {
 		t.Fatal(err)
+	}
+}
+
+// TestRangeResponseCacheStoredError verifies that if a cached entry contains
+// an error, that error is returned.
+func TestRangeResponseCacheStoredError(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	cmdID := proto.ClientCmdID{WallTime: 1, Random: 1}
+
+	// Write an error into the response cache.
+	errorReply := &proto.ReadWriteCmdResponse{}
+	errorReply.Increment = &proto.IncrementResponse{}
+	errorReply.Increment.Header().SetGoError(errors.New("boom"))
+	key := keys.ResponseCacheKey(tc.rng.Desc().RaftID, &cmdID)
+	if err := engine.MVCCPutProto(tc.engine, nil, key, proto.ZeroTimestamp, nil, errorReply); err != nil {
+		t.Fatal(err)
+	}
+
+	args, reply := incrementArgs([]byte("a"), 1, 1, tc.store.StoreID())
+	args.CmdID = cmdID
+	err := tc.rng.AddCmd(tc.rng.context(), proto.Call{Args: args, Reply: reply}, true)
+	if err == nil {
+		t.Fatal("expected to see cached error but got nil")
+	} else if ge, ok := err.(*proto.Error); !ok {
+		t.Fatalf("expected proto.Error but got %s", err)
+	} else if !reflect.DeepEqual(ge, &proto.Error{Message: "boom"}) {
+		t.Fatalf("expected error message 'boom' but got %s", ge)
 	}
 }
 

